### PR TITLE
[Go] Address Gosec vulnerabilities in Go client mustache template

### PR DESCRIPTION
**Pull request by gcatanese**
@antihax @grokify @kemokemo @jirikuncar @ph4r5h4d 

Updated Go client mustache template to address gosec warnings, updated the samples too

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.0.1) (patch release), `6.1.x` (breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
@gcatanese thanks for the PR. CI failed with the following:
```
Running node 2 to test Go
go version go1.17.6 linux/amd64
--2022-06-05 10:14:10--  https://dl.google.com/go/go1.14.linux-amd64.tar.gz
Resolving dl.google.com (dl.google.com)... 172.253.115.190, 172.253.115.93, 172.253.115.91, ...
Connecting to dl.google.com (dl.google.com)|172.253.115.190|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 123550266 (118M) [application/gzip]
Saving to: ‘STDOUT’

-                   100%[===================>] 117.83M  42.7MB/s    in 2.8s    

2022-06-05 10:14:13 (42.7 MB/s) - written to stdout [123550266/123550266]

go version go1.14 linux/amd64
go: found github.com/stretchr/testify/assert in github.com/stretchr/testify v1.7.1
go: github.com/stretchr/testify upgrade => v1.7.1
go: golang.org/x/oauth2 upgrade => v0.0.0-20220524215830-622c5d57e401
go: found golang.org/x/net/context in golang.org/x/net v0.0.0-20220531201128-c960675eff93
go: golang.org/x/net upgrade => v0.0.0-20220531201128-c960675eff93
# github.com/OpenAPITools/openapi-generator/samples/client/petstore/go/go-petstore
go-petstore/client.go:424:9: no new variables on left side of :=
FAIL	github.com/OpenAPITools/openapi-generator/samples/client/petstore/go [build failed]
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (go-test) on project GoPetstore: Command execution failed.: Process exited with an error: 2 (Exit value: 2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
```

can you please take a look when you've time?

ref: https://app.circleci.com/pipelines/github/OpenAPITools/openapi-generator/17692/workflows/744c8de9-bdba-4ac3-8281-001aa187754f/jobs/40454

**Comment by gcatanese** [comment]
@wing328 This has been fixed (apologies, I thought I had corrected already the error)

**Comment by wing328** [comment]
@gcatanese no problem. let me try to review tomorrow or over the weekend. thanks again for the PR.

File: modules/openapi-generator/src/main/resources/go/client.mustache [hunk 1]
```diff
@@ -457,11 +457,14 @@ func (c *APIClient) decode(v interface{}, b []byte, contentType string) (err err
...
-	file, err := os.Open(path)
+    file, err := os.Open(filepath.Clean(path))
 	if err != nil {
 		return err
 	}
-	defer file.Close()
+    err = file.Close()
+    if err != nil {
+        return err
+    }
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/go/client.mustache:467]: LGTM. I'll replace 4-spaces with tabs after merging this PR.

Comment (wing328) [reply modules/openapi-generator/src/main/resources/go/client.mustache:467]: Updated via 3bc44165545

**Pull request merged**